specs/sharding/beacon-chain.md

```diff
@@ -103,786 +87,558 @@ The following values are (non-configurable) constants used throughout the specif
 
-```python
-class Builder(Container):
-    pubkey: BLSPubkey
-    # TODO: fields for either an expiry mechanism (refunding execution account with remaining balance) 
-    #  and/or a builder-transaction mechanism.
-```
+    sharded_data_commitment_count: uint64 # Count of sharded data commitments
 
-### `DataCommitment`
+    bid: Gwei # Block builder bid paid to proposer
 
-```python
-class DataCommitment(Container):
-    # KZG10 commitment to the data
-    point: BLSCommitment
-    # Length of the data in samples
-    samples_count: uint64
+    validator_index: ValidatorIndex # Validator index for this bid
+    
+    # Block builders use an Eth1 address -- need signature as
+    # block builder fees and data gas base fees will be charged to this address
+	signature_y_parity: bool
+	signature_r: uint256
+	signature_s: uint256    
 ```
 
-### `AttestedDataCommitment`
+#### `IntermediateBlockBidWithRecipientAddress`
 
 ```python
-class AttestedDataCommitment(Container):
-    # KZG10 commitment to the data, and length
-    commitment: DataCommitment
-    # hash_tree_root of the ShardBlobHeader (stored so that attestations can be checked against it)
-    root: Root
-    # The proposer who included the shard-header
-    includer_index: ValidatorIndex
+class IntermediateBlockBidWithRecipientAddress(Container):
+    intermediate_block_bid: Union[None, IntermediateBlockBid]
+    ethereum_address: Bytes[20] # Address to receive the block builder bid
 ```
 
-### `ShardBlobBody`
-
-Unsigned shard data, bundled by a shard-builder.
-Unique, signing different bodies as shard proposer for the same `(slot, shard)` is slashable.
+#### `ShardedCommitmentsContainer`
 
 ```python
-class ShardBlobBody(Container):
-    # The actual data commitment
-    commitment: DataCommitment
-    # Proof that the degree < commitment.samples_count * POINTS_PER_SAMPLE
-    degree_proof: BLSCommitment
-    # The actual data. Should match the commitment and degree proof.
-    data: List[BLSPoint, POINTS_PER_SAMPLE * MAX_SAMPLES_PER_BLOB]
-    # fee payment fields (EIP 1559 like)
-    # TODO: express in MWei instead?
-    max_priority_fee_per_sample: Gwei
-    max_fee_per_sample: Gwei
-```
-
-### `ShardBlobBodySummary`
+class ShardedCommitmentsContainer(Container):
+    sharded_commitments: List[KZGCommitment, 2 * MAX_SHARDS]
 
-Summary version of the `ShardBlobBody`, omitting the data payload, while preserving the data-commitments.
+    # Aggregate degree proof for all sharded_commitments
+    degree_proof: KZGCommitment
 
-The commitments are not further collapsed to a single hash, 
-to avoid an extra network roundtrip between proposer and builder, to include the header on-chain more quickly.
+    # The sizes of the blocks encoded in the commitments (last intermediate and all beacon blocks since)
+    included_block_sizes: List[uint64, MAX_BEACON_BLOCKS_BETWEEN_INTERMEDIATE_BLOCKS + 1]
```

Comment:
`MAX_BEACON_BLOCKS_BETWEEN_INTERMEDIATE_BLOCKS` is undefined in this spec (might be OK if it's a PoC).